### ISSUE TITLE
Add new `ca_certificate` parameter

### ISSUE THREAD

**Issue by alexjfisher** [voxpupuli/puppet-archive#188]
As an alternative to setting `allow_insecure` to `true`, some users might prefer to specify their own cacerts file.

For `wget` the option is `--ca-certificate`
For `curl` the option is `--with-ca-bundle`
For `ruby` `http.ca_file`?

**Comment by igalic** [comment]
imo, you should probably ~*~just~*~ manage your ca stores "correctly" and have your CAs in (***all***) ca stores

**Comment by alexjfisher** [comment]
I've taken a closer look.  The ruby provider currently sets the `SSL_CERT_FILE` environment variable to the cacerts.pem file the module ships, but only for Windows and only if that environment variable was not previously set.  Being able to use a user provided file would be an improvement here.  I'm not sure what the history is and how/why Windows differs from Linux.

For curl/wget, I guess ideally, users would always have all their ca stores setup correctly, but providing this functionality might still be useful to some?  If I find a bit of spare time, implementing this feature for these providers should be fairly trivial.  Perhaps somebody later could then fix up the ruby provider and probably implement `allow_insecure` at the same time.

**Comment by nanliu** [comment]
TLDR; this was to work around a cert issue in some version of Windows Ruby. This was never intended as a feature.

Specifically this issue: https://gist.github.com/luislavena/f064211759ee0f806c88

Basically, a version of Windows ruby shipped a cacert that wouldn't even connect to rubygems.org. When this module had gem dependencies, this work around was required to bootstrap itself. At this point, we no longer have gem dependencies and hopefully the Ruby packages on windows does not suffer from that bug anymore, so we should be able to remove this. I agree with @igalic that the cacerts should be handled by the system and not embedded in a puppet module.

**Comment by LTangaF** [comment]
I actually recently encountered this bug and have a PR that I'm going to be creating here on that.  The environment variable SSL_CERT_FILE on Windows seems to be being ignored by Net::HTTP and I've coded a workaround that sets it specifically on the http instance before 'start'ing it.  Puppet's shipped environment.bat seems to set that environment variable to a non-existing file in C:\Windows\System32\ssl by default and we've had to make sure that file was created, too.
